src/main/java/com/google/cloud/pontem/Util.java

```diff
@@ -93,13 +96,39 @@
     return options;
   }
 
+  /** Fetch a list of all database names. */
+  public static ImmutableList<String> getListOfDatabaseNames(
+      String projectId, String instanceId, int numDatabases) {
+    LOG.info("Begin getting list of database names");
+    SpannerOptions options = Util.getSpannerOptionsBuilder().build();
+    Spanner spanner = options.getService();
+
+    DatabaseAdminClient dbAdminClient = spanner.getDatabaseAdminClient();
+
+    List<String> databaseNames = new ArrayList<>();
+
+    try {
```

Comment:
Can you use try with resources and have it close Spanner or it doesn't implement the interface?

Reply:
I don't understand your question.

Reply:
If the Spanner interface implements closeable you could do

try (Spanner spanner = options.getService()) {
  Your code
}

While avoiding the finally statement.
See: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
I do not believe Spanner implements the Closable interface.
You can disregard the E2E tests failing. I've run repeatedly locally and I'm trying to figure out what is up with Travis environment.